sdk/core/azure-core/src/logging/logging.cpp

```diff
@@ -7,72 +7,41 @@
 #include <mutex>
 
 using namespace Azure::Core::Logging;
-using namespace Azure::Core::Logging::Details;
-
-class Azure::Core::Logging::Details::LogClassificationsPrivate {
-public:
-  static LogClassifications const LogClassificationsConstant(bool all)
-  {
-    return LogClassifications(all);
-  }
-
-  static bool IsClassificationEnabled(LogClassifications const& cls, LogClassification const& c)
-  {
-    return cls.m_all || (cls.m_classifications.find(c) != cls.m_classifications.end());
-  }
-};
+using namespace Azure::Core::Logging::Internal;
 
 namespace {
 std::mutex g_loggerMutex;
```

Comment:
I think you could kill the mutex and just make the callback function atomic. This would make the code smaller and faster.

Reply:
`std::function` is not atomic, and also we want to lock both the level and the callback, or otherwise you may get unexpected levels passed to your your old/new callback.

Reply:
We generally don't are about unexpected levels being passed to the callback - the callback HAS to be written to expect any level at any time.

Reply:
Unfortunately, `std::atomic<std::function<...>>` does not compile, underlying type does not correspond to atomic requirements.